cluster/gce/util.sh

```diff
@@ -30,13 +30,30 @@ else
   exit 1
 fi
 
+function get_latest_gci_image() {
+  # GCI milestone to use
+  milestone="53"
+
+  # First try to find an actie (non-deprecated) image on this milestone.
+  gci_images=( $(gcloud compute images list --project google-containers \
+      --no-standard-images --sort-by="~creationTimestamp" \
+      --regexp="gci-[a-z]+-${milestone}-.*" --format="table[no-heading](name)") )
+
+  # If no active image is available, search across all deprecated images.
+  if [[ ${#gci_images[@]} == 0 ]] ; then
+    gci_images=( $(gcloud compute images list --project google-containers \
+        --no-standard-images --show-deprecated --sort-by="~creationTimestamp" \
+        --regexp="gci-[a-z]+-${milestone}-.*" --format="table[no-heading](name)") )
+  fi
+
+  echo "${gci_images[0]}"
+}
+
 if [[ "${OS_DISTRIBUTION}" == "gci" ]]; then
-  # If the master image is not set, we use the latest GCI dev image.
+  # If the master image is not set, we use the pinned GCI image.
   # Otherwise, we respect whatever is set by the user.
-  gci_images=( $(gcloud compute images list --project google-containers \
-      --show-deprecated --no-standard-images --sort-by='~creationTimestamp' \
-      --regexp='gci-[a-z]+-53-.*' --format='table[no-heading](name)') )
-  MASTER_IMAGE=${KUBE_GCE_MASTER_IMAGE:-"${gci_images[0]}"}
+  gci_image=$(get_latest_gci_image)
+  MASTER_IMAGE=${KUBE_GCE_MASTER_IMAGE:-"${gci_image}"}
```

Comment:
Can you collapse these two lines? I think doing so would skip the call to the get_latest_gci_image function in the case where the image is specified, which would be nice.